cmd/kube-apiserver/app/server.go

```diff
@@ -275,7 +245,7 @@ func Run(s *options.APIServer) error {
 	genericConfig.SupportsBasicAuth = len(s.BasicAuthFile) > 0
 	genericConfig.Authorizer = authorizer
 	genericConfig.AdmissionControl = admissionController
-	genericConfig.APIResourceConfigSource = apiResourceConfigSource
+	genericConfig.APIResourceConfigSource = storageFactory.APIResourceConfigSource
```

Comment:
Just a note here.  While running openshift with multiple API versions, we found that we wanted to support skewed storage encodings.  This let us store data in the new format, while exposing the REST API through the old version for old clients.  By setting these to same value, you won't be able to have disparate rules for allowed storage encodings versus REST API exposure.